Tests/test_iterator.py

```diff
@@ -256,4 +275,13 @@ def __next__(self):
         a = A()
         self.assertEqual(next(a), 2)
 
+    def test_method_changing_iterator(self):
+        switcher = IteratorSwitcher()
```

Comment:
It's the same code path. I used a for loop here because I wanted to put a safety valve on the test -- if someone accidentally reversed the behavior then the test will run for a long time before presumably running out of memory.